post_installation_configuration/preparing-for-users.adoc

```diff
@@ -0,0 +1,145 @@
+|Configure the `keystone` identity provider to integrate
+your {product-title} cluster with Keystone to enable shared authentication with
+an OpenStack Keystone v3 server configured to store users in an internal
+database.
+
+|xref:../authentication/identity_providers/configuring-ldap-identity-provider.adoc#configuring-ldap-identity-provider[LDAP]
+|Configure the `ldap` identity provider to validate user names and passwords
+against an LDAPv3 server, using simple bind authentication.
+
+|xref:../authentication/identity_providers/configuring-basic-authentication-identity-provider.adoc#configuring-basic-authentication-identity-provider[Basic authentication]
+|Configure a `basic-authentication` identity provider for users to log in to
+{product-title} with credentials validated against a remote identity provider.
+Basic authentication is a generic backend integration mechanism.
+
+|xref:../authentication/identity_providers/configuring-request-header-identity-provider.adoc#configuring-request-header-identity-provider[Request header]
+|Configure a `request-header` identity provider to identify users from request
+header values, such as `X-Remote-User`. It is typically used in combination with
+an authenticating proxy, which sets the request header value.
+
+|xref:../authentication/identity_providers/configuring-github-identity-provider.adoc#configuring-github-identity-provider[GitHub or GitHub Enterprise]
+|Configure a `github` identity provider to validate user names and passwords
+against GitHub or GitHub Enterprise's OAuth authentication server.
+
+|xref:../authentication/identity_providers/configuring-gitlab-identity-provider.adoc#configuring-gitlab-identity-provider[GitLab]
+|Configure a `gitlab` identity provider to use
+link:https://gitlab.com/[GitLab.com] or any other GitLab instance as an identity
+provider.
+
+|xref:../authentication/identity_providers/configuring-google-identity-provider.adoc#configuring-google-identity-provider[Google]
+|Configure a `google` identity provider using
+link:https://developers.google.com/identity/protocols/OpenIDConnect[Google's OpenID Connect integration].
+
+|xref:../authentication/identity_providers/configuring-oidc-identity-provider.adoc#configuring-oidc-identity-provider[OpenID Connect]
+|Configure an `oidc` identity provider to integrate with an OpenID Connect
+identity provider using an
+link:http://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth[Authorization Code Flow].
+
+|===
+
+Once an identity provider has been defined, you can
+xref:../authentication/using-rbac.adoc#authorization-overview_using-rbac[use RBAC to define and apply permissions].
+
+include::modules/identity-provider-parameters.adoc[leveloffset=+1]
+
+include::modules/identity-provider-default-CR.adoc[leveloffset=+1]
+
+[id="post-install-using-rbac-to-define-and-apply-permissions"]
+== Using RBAC to define and apply permissions
```

Comment:
`You can manually modify the default cluster roles, if required, but you must take extra steps each time you restart a master node.`
I think there is no ` restart a master node` action now,we can change this to `but you must take extra steps each time you upgrade your cluster`

Reply:
Under file **modules/rbac-overview.adoc**